packages/artifact/src/internal-download-http-client.ts

```diff
@@ -0,0 +1,132 @@
+import * as fs from 'fs'
+import {
+  createHttpClient,
+  getArtifactUrl,
+  getRequestOptions,
+  isSuccessStatusCode,
+  isRetryableStatusCode
+} from './internal-utils'
+import {URL} from 'url'
+import {
+  ListArtifactsResponse,
+  QueryArtifactResponse
+} from './internal-contracts'
+import {IHttpClientResponse} from '@actions/http-client/interfaces'
+import {HttpClient} from '@actions/http-client'
+import {DownloadItem} from './internal-download-specification'
+import {getDownloadFileConcurrency} from './internal-config-variables'
+import {warning} from '@actions/core'
+
+/**
+ * Gets a list of all artifacts that are in a specific container
+ */
+export async function listArtifacts(): Promise<ListArtifactsResponse> {
+  const artifactUrl = getArtifactUrl()
+  const client = createHttpClient()
+  const requestOptions = getRequestOptions('application/json')
+
+  const rawResponse = await client.get(artifactUrl, requestOptions)
+  const body: string = await rawResponse.readBody()
+  if (isSuccessStatusCode(rawResponse.message.statusCode) && body) {
+    return JSON.parse(body)
+  } else {
+    // eslint-disable-next-line no-console
+    console.log(rawResponse)
+    throw new Error(`Unable to list artifacts for the run`)
+  }
+}
+
+/**
+ * Fetches a set of container items that describe the contents of an artifact
+ * @param artifactName the name of the artifact
+ * @param containerUrl the artifact container URL for the run
+ */
+export async function getContainerItems(
+  artifactName: string,
+  containerUrl: string
+): Promise<QueryArtifactResponse> {
+  // The itemPath search parameter controls which containers will be returned
+  const resourceUrl = new URL(containerUrl)
+  resourceUrl.searchParams.append('itemPath', artifactName)
+
+  const client = createHttpClient()
+  const rawResponse = await client.get(resourceUrl.toString())
+  const body: string = await rawResponse.readBody()
+  if (isSuccessStatusCode(rawResponse.message.statusCode) && body) {
+    return JSON.parse(body)
+  }
+  throw new Error(`Unable to get ContainersItems from ${resourceUrl}`)
```

Comment:
You have different styles within the same file.  :)  Why not log the raw response here too like you did on line 34?

Reply:
Updated!